ORSSample/main.py

```diff
@@ -0,0 +1,130 @@
+import os
+import time
+from settings import *
+from azure.storage.blob import BlobServiceClient
+
+source_connect_str = SOURCE_CONNECTION_STRING
+dest_connect_str = DESTINATION_CONNECTION_STRING
+source_container_name = SOURCE_CONTAINER_NAME
+dest_container_name = DESTINATION_CONTAINER_NAME
+blobs = BLOB_NAMES
+
+
+def main():
+
+    source_blob_service_client = BlobServiceClient.from_connection_string(source_connect_str)
+    source_blob_container_client = source_blob_service_client.get_container_client(source_container_name)
+    source_blob_client = source_blob_service_client.get_blob_client(container=source_container_name, blob=blobs[0])
+
+    print("Uploading source blobs...")
+    upload_blobs(source_blob_service_client, source_container_name, blobs)
+
+    print("Replicating...")
+    # check_completion(source_blob_client)
+    track_progress(source_blob_service_client, source_container_name, blobs)
+    print("Replication Complete")
+
+    dest_blob_service_client = BlobServiceClient.from_connection_string(dest_connect_str)
+    dest_blob_client = dest_blob_service_client.get_blob_client(container=dest_container_name, blob=blobs[0])
+
+    print_contents(source_blob_client, dest_blob_client)
+
+    print("Updating source blob...")
+    source_blob_client.upload_blob("Lorem Ipsum", overwrite=True)
+
+    print("Replicating...")
+    check_completion(source_blob_client)
+    print("Replication Complete")
+    print_contents(source_blob_client, dest_blob_client)
+
+    print("Archiving...")
+    archive_blobs(source_blob_container_client, blobs)
+    # archive_batch(source_blob_container_client, blobs)
+    print("Archived")
+
+
+# Upload multiple blobs
+def upload_blobs(source_blob_service_client, container_name, blobs):
+    for x in blobs:
+        source_blob_client = source_blob_service_client.get_blob_client(container=container_name, blob=x)
+        source_blob_client.upload_blob("Hello World!", overwrite=True)
+
+
+# Tracks progress of replicating multiple blobs
+def track_progress(source_blob_service_client, container_name, blobs):
+    completed = 0
+    length = len(blobs)
+    incomplete_events = blobs.copy()
+    completed_events = []
+
+    while completed < length:
+        for x in incomplete_events:
+            blob_client = source_blob_service_client.get_blob_client(container=container_name, blob=x)
+            props = blob_client.get_blob_properties()
+
+            for replication_policy in props.object_replication_source_properties:
+                for rule in replication_policy.rules:
+                    if rule.status == "complete":
+                        completed += 1
+                        completed_events.append(x)
+                        incomplete_events.remove(x)
```

Comment:
It's probably best if you don't modify what you're iterating over. So instead, maybe do "for incomplete_event in incomplete_events[:]:" so you iterate over a copy of the original list